kratos/utilities/time_discretization.h

```diff
@@ -0,0 +1,113 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+#if !defined(KRATOS_TIME_DISCRETIZATION_H_INCLUDED )
+#define  KRATOS_TIME_DISCRETIZATION_H_INCLUDED
+
+// System includes
+#include <vector>
+
+// External includes
+
+// Project includes
+
+namespace Kratos {
+namespace TimeDiscretization {
+
+class BDF1
+{
+public:
+    void ComputeBDFCoefficients(const double DeltaTime,
+                                std::vector<double>& rCoefficients) const;
+};
+
+class BDF2
+{
+public:
+    void ComputeBDFCoefficients(const double DeltaTime,
+                                std::vector<double>& rCoefficients) const;
```

Comment:
```suggestion
                                std::array<double, 3>& rCoefficients) const;
```
Similar changes for the others since the size is known at compile time.

Reply:
@msandre you need to pass this to the processinfo as a variable, so i think it must be a variable-size vector

Reply:
actually this is exactly the point where we disagree

I want to keep those classes lightweight, if someone needs the BDF_COEFFICIENTS in the processinfo then they can do it themselves

=> I am rather sure that this won't even be necessary, I think the original reason this was done bcs this is a way to return values from a process

Reply:
It seems like all the code for computing BDF_COEFFICIENTS is the same and boils down to:

```
//calculate the BDF coefficients
            double Dt = rCurrentProcessInfo[DELTA_TIME];
            double OldDt = rCurrentProcessInfo.GetPreviousTimeStepInfo(1)[DELTA_TIME];

            double Rho = OldDt / Dt;
            double TimeCoeff = 1.0 / (Dt * Rho * Rho + Dt * Rho);

            Vector& BDFcoeffs = rCurrentProcessInfo[BDF_COEFFICIENTS];
            BDFcoeffs.resize(3, false);

            BDFcoeffs[0] = TimeCoeff * (Rho * Rho + 2.0 * Rho); //coefficient for step n+1 (3/2Dt if Dt is constant)
            BDFcoeffs[1] = -TimeCoeff * (Rho * Rho + 2.0 * Rho + 1.0); //coefficient for step n (-4/2Dt if Dt is constant)
            BDFcoeffs[2] = TimeCoeff; //coefficient for step n-1 (1/2Dt if Dt is constant)
```

Error checking on buffer and time step sizes tends to vary more. We could replace these with something like

```
#include "utilities/time_discretization.h"
...
std::array<double, 3> a;
BDF2(Dt,OldDt).ComputeBDFCoefficients(a);
BDFcoeffs = {a.begin(), a.end()};
```